### PR TITLE
feat: rewrite if-to-if

### PR DESCRIPTION
**Pull request by ChrBondi**
rewrite if-to-if #3638

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Inliner.scala [hunk 1]
```diff
@@ -230,10 +230,28 @@ object Inliner {
       val e1 = visitExp(exp1, subst0)
       val e2 = visitExp(exp2, subst0)
       val e3 = visitExp(exp3, subst0)
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Inliner.scala:232]: Usually names like e1, e2, and e3 are fine when things are simple. That is not the case here. May I propose:

`outerCond`, `outerThen`, `outerElse`, `innerCond`, `innerThen`, and `innerElse` :)

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Inliner.scala:232]: Also, did you consider if this should also happen inside substituteExp or not?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Inliner.scala:232]: (Not saying it has to, just if you considered)

Comment (ChrBondi) [reply main/src/ca/uwaterloo/flix/language/phase/Inliner.scala:232]: I think it's a good idea to have it both places. Seems like 1/4 of the nested ifs are being seen in substituteExp after a quick measurement

**Comment by magnus-madsen** [comment]
Looks good, but there is a conflict.

**Pull request merged**